### DISCUSSION TITLE
Quorum queue per message expiration (again)

### DISCUSSION THREAD

**Discussion by AlexBenoitUbi** [rabbitmq/rabbitmq-server#10618] [Other] [answered]
### Describe the bug

Per the RabbitMQ [documentation](https://www.rabbitmq.com/docs/ttl#message-ttl-dead-lettering), the new Quorum queues should handle per message expiration as it happens instead of waiting for the message to get to the head of the queue.

I've made a setup to test this with the following arguments on my queue:
```c#
var QuorumQueueArguments = new Dictionary<string, object>
{
    { "x-dead-letter-exchange", "MyExchangeName" },
    { "x-queue-type", "quorum" }
};
```
From my manual testing, the behavior is the same as the normal queues. It waits for the message to get to the head of the queue. Am I missing something or is it a bug?

### Reproduction steps

1. Create a quorum queue
2. Send a message with a long expiration
3. Send a second message with a shorter expiration

### Expected behavior

The expected behavior is that the second message with a shorter expiration should be "dead-lettered" before the first message with a long expiration before it gets to the head of the queue as mentioned in the documentation. Instead, we are observing the same behavior as the normal queues.

**Comment by michaelklishin** [comment, marked as answer]
This is the expected behavior, the docs need to be corrected.

**Comment by michaelklishin** [comment]
This was discussed to dead in https://github.com/rabbitmq/rabbitmq-server/discussions/10516 but the docs were updated mid-way and not at the end of that discussion.
